### PR TITLE
build: auto-push patch file changes

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Small bit of code to auto push three way merge patch file changes from CI.  Abuses a GH app to do it similar to how trop works

Notes: no-notes

### CODE REVIEW

File: patches/chromium/rename_the_v8_context_snapshot_on_arm64_macos_builds.patch [hunk 1]
```diff
@@ -22,7 +22,7 @@ index 49ba7977cc0d5710918b3452ab57078ba31c9949..a362123cad17d061690e7cb5a26413fd
      arch_suffix = "32"
      if (use_v8_context_snapshot) {
 -      renaming_sources = [ "$_secondary_abi_out_dir/v8_context_snapshot.bin" ]
-+      renaming_sources = [ "$_secondary_abi_out_dir/$v8_context_snapshot_filename" ]
++      renaming_sources = [ "$_secondary_abi_out_dir/$v8_context_snapshot_filename"]
```
Comment (nornagon) [inline patches/chromium/rename_the_v8_context_snapshot_on_arm64_macos_builds.patch:25]: good change but unrelated

File: script/push-patch.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  privateKey: Buffer.from(process.env.PATCH_UP_PRIVATE_KEY, 'base64').toString('utf8'),
+  installationId: process.env.PATCH_UP_INSTALLATION_ID,
+  clientId: process.env.PATCH_UP_CLIENT_ID,
+  clientSecret: process.env.PATCH_UP_CLIENT_SECRET
+});
+
+async function main () {
+  const installationAuth = await auth({ type: 'installation' });
+  const remoteURL = `https://x-access-token:${installationAuth.token}@github.com/electron/electron.git`;
+  console.log(remoteURL);
```
Comment (nornagon) [inline script/push-patch.js:25]:
```suggestion
```

**Review by nornagon — APPROVED** [review]
noice

**Review by jkleinsc — APPROVED** [review]
LGTM -- just needs linting.

**Review by ckerr — APPROVED** [review]
:tada:

**Comment by jkleinsc** [comment]
Merging as Mac failure is flake unrelated to this PR.

**Pull request merged**